.pre-commit-config.yaml

```
@@ -4,48 +4,74 @@ ci:
 exclude: ".*\\.asdf$"
 
 repos:
+  # basic checks
   - repo: https://github.com/pre-commit/pre-commit-hooks
     rev: v6.0.0
     hooks:
       - id: check-added-large-files
       - id: check-ast
       - id: check-case-conflict
       - id: check-yaml
-        args: ["--unsafe"]
+        args:
+          - --unsafe
       - id: check-toml
       - id: check-merge-conflict
       - id: check-symlinks
-      - id: debug-statements
-        exclude: "src/stpipe/cmdline.py"
+      # - id: debug-statements
       - id: detect-private-key
       - id: end-of-file-fixer
       - id: trailing-whitespace
-
+  # simple text searches for common issues
   - repo: https://github.com/pre-commit/pygrep-hooks
     rev: v1.10.0
     hooks:
-      - id: python-check-blanket-noqa
-      - id: python-check-mock-methods
       - id: rst-directive-colons
       - id: rst-inline-touching-normal
       - id: text-unicode-replacement-char
-
+  # lint and format
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.12.10
+    hooks:
+      - id: ruff-check
+        args:
+          - --fix
+          - --show-fixes
+      # - id: ruff-format
```

```suggestion
      # - id: ruff-format
```
Don't we need this to replace the removed black pre-commit hook?

---

Yeah, good point. I commented it out at first because it was changing files, but it looks like it only  changes two small lines (I'm not sure why black wasn't fixing those)